app/controllers/addFinancialInstitution/WhichIdentificationNumbersController.scala

```diff
@@ -70,9 +70,15 @@ class WhichIdentificationNumbersController @Inject() (
           value =>
             for {
               updatedAnswers <- Future.fromTry(request.userAnswers.set(WhichIdentificationNumbersPage, value))
-              _              <- sessionRepository.set(updatedAnswers)
-              _              <- changeUserAnswersRepository.set(request.fatcaId, updatedAnswers.get(ChangeFiDetailsInProgressId), updatedAnswers)
-            } yield Redirect(navigator.nextPage(WhichIdentificationNumbersPage, mode, updatedAnswers))
+              cleanedAnswers <- Future.fromTry(
+                WhichIdentificationNumbersPage.cleanUpUnselectedTINPages(
+                  selectedTINs = value.map(_.toString),
```

Comment:
instead of using `_.toString`, we can use the type `Set[WhichIdentificationNumbers]` as is